File: DemoApp Objective-C/CASSample/RewardedAd/RewardedVC.m
//
//  RewardedVC.m
//  CASSample
//

#import "RewardedVC.h"
#import "AppDelegate.h"

@interface RewardedVC ()

@property (strong, nonatomic) CASRewarded *rewarded;

@end

@implementation RewardedVC

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.rewarded = [[CASRewarded alloc] initWithCasID: [AppDelegate casId]];
    self.rewarded.delegate = self;
    self.rewarded.impressionDelegate = self;
    
    self.navigationItem.largeTitleDisplayMode = UINavigationItemLargeTitleDisplayModeNever;
    self.navigationItem.title = @"Rewarded Ad";
}

- (IBAction)loadAction:(UIButton *)sender {
    [self.rewarded loadAd];
}

- (IBAction)showAction:(UIButton *)sender {
    [self.rewarded presentFromViewController:self userDidEarnRewardHandler: ^(id rewarded) {
        NSLog(@"userDidEarnRewardHandler");
    }];
}

#pragma mark - CASImpressionDelegate

- (void)adDidRecordImpressionWithInfo:(CASContentInfo *)info {
    NSLog(@"adDidRecordImpressionWithInfo");
}

#pragma mark - CASScreenContentDelegate

- (void)screenAdDidLoadContent:(id<CASScreenContent>)ad {
    NSLog(@"screenAdDidLoadContent");
}

- (void)screenAd:(id<CASScreenContent>)ad didFailToLoadWithError:(CASError *)error {
    NSLog(@"screenAd:didFailToLoadWithError %@", error.description);
}

- (void)screenAdWillPresentContent:(id<CASScreenContent>)ad {
    NSLog(@"screenAdWillPresentContent");
}

- (void)screenAd:(id<CASScreenContent>)ad didFailToPresentWithError:(CASError *)error {
    NSLog(@"screenAd:didFailToPresentWithError %@", error.description);
}

- (void)screenAdDidClickContent:(id<CASScreenContent>)ad {
    NSLog(@"screenAdDidClickContent");
}

- (void)screenAdDidDismissContent:(id<CASScreenContent>)ad {
    NSLog(@"screenAdDidDismissContent");
}

@end

File: DemoApp Swift/CASSample/RewardedAd/RewardedVC.swift
//
//  RewardedVC.swift
//  CASSample
//
//  Copyright © 2025 Clever Ads Solutions. All rights reserved.
//

import UIKit
import CleverAdsSolutions

class RewardedVC: UIViewController {
    
    @IBOutlet var loadButton: UIButton!
    @IBOutlet var showButton: UIButton!
    
    private var rewarded = CASRewarded(casID: AppDelegate.casId)
        
    override func viewDidLoad() {
        super.viewDidLoad()
        rewarded.delegate = self
        rewarded.impressionDelegate = self
        //
        navigationItem.largeTitleDisplayMode = .never
        navigationItem.title = "Rewarded Ad"
    }
    
    @IBAction private func loadAction(_ sender: UIButton) {
        rewarded.loadAd()
    }
    
    @IBAction private func showAction(_ sender: UIButton) {
        rewarded.present(from: self, userDidEarnRewardHandler: { _ in
            print(#function, "Reward earned")
        })
    }
}

extension RewardedVC: CASImpressionDelegate {
    func adDidRecordImpression(info: AdContentInfo) {
        print(#function, "Ad Source: \(info.sourceName)")
    }
}

extension RewardedVC: CASScreenContentDelegate {
    func screenAdDidLoadContent(_ ad: any CASScreenContent) {
        print(#function)
    }

    func screenAd(_ ad: any CASScreenContent, didFailToLoadWithError error: AdError) {        
        print(#function, "Error: \(error.description)")
    }

    func screenAdWillPresentContent(_ ad: any CASScreenContent) {
        print(#function)
    }

    func screenAd(_ ad: any CASScreenContent, didFailToPresentWithError error: AdError) {
        print(#function, "Error: \(error.description)")
    }

    func screenAdDidClickContent(_ ad: any CASScreenContent) {
        print(#function)
    }

    func screenAdDidDismissContent(_ ad: any CASScreenContent) {
        print(#function)
    }
}
